src/main/scala/org/constellation/domain/healthcheck/ReconciliationRound.scala

```diff
@@ -0,0 +1,244 @@
+package org.constellation.domain.healthcheck
+
+import cats.effect.concurrent.Ref
+import cats.effect.{Blocker, Clock, Concurrent, ContextShift, Fiber, Timer}
+import cats.syntax.all._
+import io.chrisdavenport.log4cats.SelfAwareStructuredLogger
+import io.chrisdavenport.log4cats.slf4j.Slf4jLogger
+import io.circe.Codec
+import io.circe.generic.semiauto.deriveCodec
+import org.constellation.infrastructure.p2p.{ClientInterpreter, PeerResponse}
+import org.constellation.p2p.Cluster
+import org.constellation.schema.NodeState.{canActAsJoiningSource, isInitiallyJoining, isInvalidForJoining}
+import org.constellation.schema.consensus.RoundId
+import org.constellation.schema.{Id, NodeState}
+
+import scala.concurrent.duration.{DurationInt, FiniteDuration}
+
+class ReconciliationRound[F[_]](
+  ownId: Id,
+  cluster: Cluster[F],
+  val startedAtSecond: Long
+)(
+  apiClient: ClientInterpreter[F],
+  unboundedBlocker: Blocker
+)(implicit F: Concurrent[F], CS: ContextShift[F], C: Clock[F], T: Timer[F]) {
+  import ReconciliationRound._
+
+  val logger: SelfAwareStructuredLogger[F] = Slf4jLogger.getLogger[F]
+
+  val retryAttempts = 2
+  val retryAfter: FiniteDuration = 60.seconds
+
+  private[healthcheck] val reconciliationResult: Ref[F, Option[ReconciliationResult]] =
+    Ref.unsafe(None)
+
+//  private[healthcheck] val firstClusterAlignmentResults: Ref[F, Option[Map[Id, List[ClusterAlignmentResult]]]] =
+//    Ref.unsafe(None)
+
+//  private[healthcheck] val lastClusterAlignmentResults: Ref[F, Option[Map[Id, List[ClusterAlignmentResult]]]] =
+//    Ref.unsafe(None)
+
+  def getReconciliationResult(): F[Option[ReconciliationResult]] = reconciliationResult.get
+
+  def start(): F[Fiber[F, Unit]] = F.start(run())
+
+  private def run(
+    attemptsLeft: Int = retryAttempts,
+    firstClusterAlignmentResult: Option[Map[Id, List[ClusterAlignmentResult]]] = None,
+    lastClusterAlignmentResult: Option[Map[Id, List[ClusterAlignmentResult]]] = None
+  ): F[Unit] =
+    for {
+      isFirstRun <- F.delay(attemptsLeft == retryAttempts)
+      isLastRun <- F.delay(attemptsLeft == 0)
+      _ <- if (isFirstRun) F.unit else T.sleep(retryAfter)
```

Comment:
make sure you are performing sleep on unbounded thread pool, not blocking any other pool (!)

Reply:
Sure! Though I'm not sure if sleep does blocking at all?

Reply:
yes, sleep executes on some thread pool. either the thread pool on which it's running or a thread pool from which the `T` was created. imo needs verifying

Reply:
Verified and adjusted it today. It will run on unboundedHealth thread pool.